File: 06_strings/06_02_reverse_words_in_a_string/reverse_words_in_a_string.py
def reverse_words(s: str):
    array = s.strip().split()
    return ' '.join(array[::-1])

# Test case 1: Basic example
input1 = "Hello World"
reversed1 = reverse_words(input1)
print(f"Test Case 1: Original: \"{input1}\", Reversed: \"{reversed1}\"")

# Test case 2: Multiple spaces between words
input2 = "   OpenAI    GPT-3.5   "
reversed2 = reverse_words(input2)
print(f"Test Case 2: Original: \"{input2}\", Reversed: \"{reversed2}\"")

# Test case 3: Leading and trailing spaces
input3 = "   This is a test   "
reversed3 = reverse_words(input3)
print(f"Test Case 3: Original: \"{input3}\", Reversed: \"{reversed3}\"")

# Test case 4: Empty string
input4 = ""
reversed4 = reverse_words(input4)
print(f"Test Case 4: Original: \"{input4}\", Reversed: \"{reversed4}\"")

# Test case 5: Single word
input5 = "Python"
reversed5 = reverse_words(input5)
print(f"Test Case 5: Original: \"{input5}\", Reversed: \"{reversed5}\"")

File: 06_strings/06_02_reverse_words_in_a_string/reverse_words_in_a_string.js
const reverseWords = (s = '') => {
    const array = s.trim().split(/\s+/);
    return array.reverse().join(" ");
}

// Test case 1: Basic example
let input1 = "Hello World";
let reversed1 = reverseWords(input1);
console.log(`Test Case 1: Original: "${input1}", Reversed: "${reversed1}"`);

// Test case 2: Multiple spaces between words
let input2 = "   OpenAI    GPT-3.5   ";
let reversed2 = reverseWords(input2);
console.log(`Test Case 2: Original: "${input2}", Reversed: "${reversed2}"`);

// Test case 3: Leading and trailing spaces
let input3 = "   This is a test   ";
let reversed3 = reverseWords(input3);
console.log(`Test Case 3: Original: "${input3}", Reversed: "${reversed3}"`);

// Test case 4: Empty string
let input4 = "";
let reversed4 = reverseWords(input4);
console.log(`Test Case 4: Original: "${input4}", Reversed: "${reversed4}"`);

// Test case 5: Single word
let input5 = "JavaScript";
let reversed5 = reverseWords(input5);
console.log(`Test Case 5: Original: "${input5}", Reversed: "${reversed5}"`);